### PR TITLE
Add GO-Term

### PR DESCRIPTION
**Pull request by vjmarteau**
Minimal Working example for GO term analysis

### CODE REVIEW

File: analysis/mwe-GO-Terms.Rmd [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+# Add ENTREZ IDs
+ensembl_to_entrez <- AnnotationDbi::select(org.Hs.eg.db, unique(dat$gene_id),
+                                           keytype = "ENSEMBL",
+                                           columns = c("ENTREZID"))
+
+dat <- dat |> inner_join(ensembl_to_entrez, by = c("gene_id" = "ENSEMBL"))
+
+universe <- unique(dat$ENTREZID)
+genes <- dat |> subset(padj < 0.05) |> pull(ENTREZID) |> unique()
+```
```
Comment (vjmarteau) [inline analysis/mwe-GO-Terms.Rmd:67]: Some of the tools need entrez ids to work. You might "loose" some of your genes during maping.

File: analysis/mwe-GO-Terms.Rmd [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+WikiPathway <- enrichWP(
+  gene = genes,
+  universe = universe,
+  organism = 'Homo sapiens',
+  pvalueCutoff = 0.05)
+
+as_tibble(WikiPathway@result)
+```
+
+## Finer control on GO term tests
```
Comment (vjmarteau) [inline analysis/mwe-GO-Terms.Rmd:98]: Not sure if that is maybe too much, but you can also have a much finer control over all parameters if you want to play around.

File: analysis/mwe-GO-Terms.Rmd [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+up_AnnoList <- lapply(up_tab$"GO.ID", 
+                      function(x) as.character(unlist(genesInTerm(object = upGOdata, whichGO = x))))
+
+up_SigList <- lapply(up_AnnoList, function(x) intersect(x, up_sigGenes))
+
+#Coerce gene list to a comma-separated vector
+up_tab$Genes <- sapply(up_SigList, paste, collapse = ",")
+
+# Cbind first 5 up 
+cbind(head(up_tab$Genes, 5))
```
Comment (vjmarteau) [inline analysis/mwe-GO-Terms.Rmd:152]: This gives you the option to actually look at the individual genes that contribute to the respective process

File: analysis/mwe-GO-Terms.Rmd [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  subset(padj < 0.05) |>
+  rownames_to_column(var = "gene_id") |>
+  pull(gene_id)
+
+dat <- results_PAS_ctrl |>
+  rownames_to_column(var = "gene_id")|>
+  subset(!gene_id %in% rm)
+
+# Define significance threshold and divide into up and down genes based on logFC
+up_genes <- dat |> subset(padj < 0.05 & dat$log2FoldChange > 0) |> pull(gene_id)
```
[the remark this answers is not in the document]
Comment (TinclG) [reply analysis/mwe-GO-Terms.Rmd:38]: I did it like this 
`ENS_sig <-  ENS_sig %>%
ENS_sig$Row.names %in% ENS_bg$Row.names`

**Pull request merged**